cmd/kube-apiserver/app/server.go

```diff
@@ -346,12 +367,12 @@ func (s *APIServer) Run(_ []string) error {
 	if err != nil {
 		return err
 	}
-	storageVersions := make(map[string]string)
-	if s.StorageVersion == "" {
-		s.StorageVersion = g.Version
+
+	storageVersions := generateStorageVersionMap(s.StorageVersion, s.StorageVersions)
+	if _, found := storageVersions[g.Group]; !found {
```

Comment:
Yes. And in the future, we should fold these logic for each group into a for loop.